packages/fxa-settings/src/lib/hooks/useWebRedirect/index.tsx

```diff
@@ -24,19 +24,25 @@ export function useWebRedirect(redirectTo: BaseIntegrationData['redirectTo']) {
   const location = useLocation();
   const ftlMsgResolver = useFtlMsgResolver();
 
-  const isValid = () =>
-    redirectTo
-      ? isAllowed(redirectTo, location.href, config.redirectAllowlist)
-      : false;
+  if (!redirectTo) {
+    return;
+  }
```

Comment:
Hmm, why change it to this when it requires us to now check if `webRedirectCheck` is defined before we call `isValid`?

Reply:
I changed it because we weren't clearly differentiating between undefined `redirectTo` and invalid `redirectTo`

Reply:
Does that matter though? We just need to know if it's valid or invalid.

Reply:
I'll go ahead and approve - I personally prefer the result from the hook always being defined and then calling `isValid()` on it to check for validity but if you feel strongly that we should check that `redirectTo` is defined or not before checking the validity then I can live with that. :P

Reply:
Part of the confusion for me was that the error message was always defined, whether or not the redirectTo was valid and/or defined. Maybe it would be better for the hook to only return the redirectTo state - undefined, valid, invalid - and not an error message. It would simplify things. We could then just have the result of the hook be `webRedirectState` 

Follow-up maybe if functionally okay as is?

Reply:
> Part of the confusion for me was that the error message was always defined

Ah yeah that's valid. Yeah I'm fine with whatever you want to do, and we can get a 1-2 pointer filed in React Cleanup to look at this.